### PR TITLE
[lxd_image_vault] Get correct info for instance with custom image

### PR DESCRIPTION
**Pull request by townsend2010**

### CODE REVIEW

File: src/platform/backends/lxd/lxd_vm_image_vault.cpp [hunk 1]
```diff
@@ -163,24 +163,44 @@ mp::VMImage mp::LXDVMImageVault::fetch_image(const FetchType& fetch_type, const
...
+                }
 
-            for (const auto& alias : info.aliases)
+                return source_image;
+            }
+            catch (const std::exception&)
             {
-                source_image.aliases.push_back(alias.toStdString());
+                // Ignore
             }
```
Comment (Saviq) [inline src/platform/backends/lxd/lxd_vm_image_vault.cpp:203]: Could we be stricter here?

**Review by Saviq — CHANGES_REQUESTED** [review]
Since we can… I wonder if we should just store the metadata with the instance image? This way, even when we remove the data from the custom image host, it'd just be there in LXD?

Don't we do the same in the default vault? Copy all the image data into the instance vault?

**Comment by townsend2010** [comment]
@Saviq,

>  I wonder if we should just store the metadata with the instance image?

We do and that is what I'm using here now to get the image info for any particular instance using a custom image.

File: src/platform/backends/lxd/lxd_vm_image_vault.cpp (hunk 1 above)
Comment (townsend2010) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp:203]: Well, `info_for_full_hash()` will throw a `std::runtime_error` when it doesn't find a matching hash, but there are multiple image hosts, so we need to ignore that exception so we can move on in the loop.  So the only better granularity we can have here is to catch the `runtime_error` explicitly, but that is quite generic in and of itself.

Comment (Saviq) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp:203]: Add a TODO, please.

Comment (townsend2010) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp:203]: A TODO for what?  Change the exception type that `info_for_full_hash()` throws?  I don't think changing this to catch `std::runtime_error` is the right thing to do.

Comment (Saviq) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp:203]: Yes, for refactoring so that `info_for_full_hash` throws specific exceptions.

Comment (townsend2010) [reply src/platform/backends/lxd/lxd_vm_image_vault.cpp:204]: Ok, I added a TODO here and in the place in `ubuntu_image_host.cpp` where it really needs to be implemented.

**Comment by Saviq** [comment]
Yups.

bors merge

**Pull request merged**